### DISCUSSION TITLE
Use a different Sampler for the same Study

### DISCUSSION THREAD

**Discussion by avm19** [optuna/optuna#4619] [Q&A] [answered]
Is it possible to use a different Sampler for the same Study?

The answer is **no**, as follows from `optuna.create_study()` signature and from [the docs](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.Study.html#optuna.study.Study.optimize):
> [Study.optimze()] is done by choosing a suitable set of hyperparameter values from a given range. Uses a sampler which implements the task of value suggestion based on a specified distribution. The sampler is specified in create_study() and the default choice for the sampler is TPE. See also TPESampler for more details on ‘TPE’.

Suppose I want to try a different optimisation algorithm. I want to keep previous complete trials, so that I don't waste time already spent on them. I imagine, I could do something like:
```
new_study = optuna.create_study(...)
new_study.add_trials(old_study.get_trials())
```
Is this a reasonable solution? Will it work? Please feel free to explain to me if and why this does not make sense, as I have little experience with Bayesian Optimisation and algorithms used in Optuna Samplers. Thanks!

**Comment by nzw0301** [comment, marked as answer]
Yes, see the two approaches proposed in https://github.com/optuna/optuna/issues/2635.
